include/bl_storage.rst

```diff
@@ -16,22 +16,31 @@ The library stores the following data:
 * Image slot sizes
 * Hashes of public keys
 * Invalidation tokens used to revoke public keys
+* :ref:`Application versions <store_app_version>`
+
 
 See :ref:`bootloader_provisioning` for more information about the provisioned data and how the bootloader uses it.
 
 You can find tests for the library at :file:`tests/subsys/bootloader/bl_storage/`.
 
-Monotonic version counter
-*************************
+.. _store_app_version:
+
+Storing version information
+***************************
+
+The bootloader storage can be used to keep track of the version of the application.
+When updating the application, the :ref:`doc_bl_validation` library checks if the new version is newer than the current version and only accepts the new image if it is.
+After updating, the application must then store the new version information in the bootloader storage.
 
-The bootloader storage can contain a monotonic version counter for the app.
-This is implemented as a series of 16 bit values, where each update to the counter is written to the next available slot.
-When reading the counter, the bootloader storage library iterates over each slot checking that each is larger than the previous, to guarantee that it is monotonic (always increases).
+This functionality is implemented as a monotonic version counter that contains a series of 16-bit integer values.
+Each update to the counter is written to the next available slot.
+When reading the counter, the bootloader storage library iterates over each slot checking that each value is larger than the previous, to guarantee that the series is monotonic (always increases).
```

Comment:
We discussed what should happen if a slot is smaller than the previous, and decided it should not fail, just ignore the offending slot.

```suggestion
When reading the counter, the bootloader storage library iterates over each slot and picks the largest value, to guarantee that the counter value is monotonic (always increases).
```

Reply:
Picks the largest value for what? And then it might not be monotonic.
So maybe:
```suggestion
When reading the counter, the bootloader storage library iterates over each slot and returns the largest value.
```

Reply:
It's monotonic from the user's perspective, i.e. via the API: If a value N is written to the counter, no value smaller than N will ever be returned by the read() function, even if an attacker gains access to place arbitrary values into the slots.

Or to put it another way: It's monotonic across time, even if it's not monotonic across the slots.

Reply:
> Picks the largest value for what?

As the result of the read operation.